src/test/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheckTest.java

```diff
@@ -0,0 +1,228 @@
+        verify(checkConfig, getPath("InputOrderedProperties.txt"), expected);
+    }
+
+    @Test
+    public void testIoException() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(OrderedPropertiesCheck.class);
+        final OrderedPropertiesCheck check = new OrderedPropertiesCheck();
+        check.configure(checkConfig);
+        final String fileName =
+                getPath("InputOrderedPropertiesCheckNotExisting.properties");
+        final File file = new File(fileName);
+        final FileText fileText = new FileText(file, Collections.emptyList());
+        try {
+            final SortedSet<LocalizedMessage> messages =
+                    check.process(file, fileText);
+            fail("Expected an CheckstyleException to be thrown : " + messages);
+        }
+        catch (CheckstyleException anCheckstyleException) {
+            final String causeMessageStart = "Unable to open file";
+            assertTrue("Unable to open file ",
+                    anCheckstyleException.getMessage().startsWith(causeMessageStart));
+        }
+    }
+
+    @Test
+    public void testWrongKeyTypeInProperties() throws Exception {
+        final Class<?> orderedPropertiesClass = Class
+                .forName("com.puppycrawl.tools.checkstyle.checks."
+                        + "OrderedPropertiesCheck$SequencedProperties");
+        final Constructor<?> constructor = orderedPropertiesClass.getDeclaredConstructor();
+        constructor.setAccessible(true);
+        final Object orderedProperties = constructor.newInstance();
+        final Method method = orderedProperties.getClass().getDeclaredMethod("put", Object.class,
+                Object.class);
+        final Object result = method.invoke(orderedProperties, 1, "value");
+        final Map<Object, Object> table = new HashMap<>();
+        final Object expected = table.put(1, "value");
+        assertEquals("Invalid result of put method", expected, result);
+
+        final Object result2 = method.invoke(orderedProperties, 1, "value");
+        final Object expected2 = table.put(1, "value");
+        assertEquals("Value should be substituted", expected2, result2);
+    }
+
+    @Test
+    public void testLineNumbers() throws Exception {
+
+        // load properties into fileText
+        final String charset = System.getProperty("file.encoding", StandardCharsets.UTF_8.name());
+        final File file = new File(getPath("InputOrderedProperties2EmptyValue.properties"));
+        final FileText fileText = new FileText(file, charset);
+
+        // access private getLineNumber
+        final Class<?> orderedPropertiesClass = Class
+                .forName("com.puppycrawl.tools.checkstyle.checks.OrderedPropertiesCheck");
+        final Constructor<?> constructor = orderedPropertiesClass.getDeclaredConstructor();
+        constructor.setAccessible(true);
+
+        final Object orderedProperties = constructor.newInstance();
```

Comment:
Constructor isn't private so why is this needed?

Reply:
Because of "I do not know". There are 12 other tests doing it in the same way. I was copying it from somewhere.
I will change it.

Reply:
Those tests are probably dealing with a private constructor. We are working towards any reflection needs to be documented why it is needed. It is better we try to work without it. We just recently started removing powermock as it is causing issues with other tests and pitest. See https://github.com/checkstyle/checkstyle/issues/6439

Reply:
Done.